src/server/search/doc_accessors.cc

```diff
@@ -104,17 +108,32 @@ SearchDocData StringMapAccessor::Serialize(const search::Schema& schema) const {
   return out;
 }
 
-struct JsonAccessor::JsonPathContainer : public jsoncons::jsonpath::jsonpath_expression<JsonType> {
+struct JsonAccessor::JsonPathContainer
+    : public variant<json::Path, jsoncons::jsonpath::jsonpath_expression<JsonType>> {
+  vector<JsonType> Evaluate(const JsonType& json) const {
+    vector<JsonType> res;
+
+    visit(Overloaded{[&](const json::Path& path) {
+                       json::EvaluatePath(path, json,
+                                          [&](auto, const JsonType& v) { res.push_back(v); });
+                     },
+                     [&](const jsoncons::jsonpath::jsonpath_expression<JsonType>& path) {
+                       auto json_arr = path.evaluate(json);
+                       for (const auto& v : json_arr.array_range())
+                         res.push_back(v);
+                     }},
+          *this);
```

Comment:
nit: if you store the overloaded separately it becomes more readable

Reply:
updated due to compile error on an older compiler